Manage Rewards (#150)
exoplatform/exo-documentation: docs/Administration.rst

	-  :ref:`Managing kudos <ManagingKudos>`
       How to manage the use of kudos by users.
	   
	-  :ref:`Managing rewards <ManagingRewards>`

>        How to manage the use of kudos by users.

Reviewer: ```suggestion
       How to manage the rewarding programs.
```

|image336|

You can filter per period type and select the period in which you want to see all sent (and recieved) kudos.

|image335|

.. _ManagingRewards:

================
Managing Rewards
================

You can manage the rewards sending from "Reward administration" page. 
To access to this interface, click on "Administration" menu from the top navigation, then "Reward" > "Reward" from the drop-down menu.
The first thing to do is to setup how rewards will be calculated from the configuration tab. 
Then, from "Pools" tab you can create pools and assign users to them according to their domains. 

> Since the use of the platform may not be the same from one team to another, and in order to ensure a fair distribution of rewards, you can divide users by pools according to their teams. 

Reviewer: ```suggestion
Since the use of the platform may not be the same from one team to another, and in order to ensure a fair distribution of rewards, regroup users in pools corresponding to their teams. 
```

> This will put the members of the same team in competition together but not with all the users of the platform.

Reviewer: ```suggestion
Rewards will be distributed comparatively to members within the same pool. Hence, members of the same pool will compete together for rewards rather than with all the users of the platform.
```

Finally, when everything is setup correctly in both configuration and pools tabs, you can send rewards from the first tab.

Configuration
~~~~~~~~~~~~~

Configuration is the third tab of Reward interface. 
This tab lets you manage the rewards sending, the periodicity and how rewards will be calculated for users at the end of each period.

|image337|

> Choose the periodicity for sending rewards. The possible values are : week, month, quarter, semester or year.

Reviewer: ```suggestion
Choose the periodicity for sending rewards  : week, month, quarter, semester or year.
```

> The interface is divided into two blocks : a block to setup the rewards for kudos and a block to setup the rewards for gamification points.

Reviewer: ```suggestion
The interface currently offers two built-in rewarding programs : one  for kudos and  one for gamification points.
```

> For each bloc you can specify a minimal threshold to reward users. The default value is "0", which means that the users are rewarded automatically if they receive kudos or earn any gamification points.

Reviewer: ```suggestion
For each program, you can specify a minimal threshold to reward users. The default value is "0", which means that the users are rewarded automatically if they receive kudos or earn any gamification points.
```

> Select the checkbox "Use pools" if you want to use the pools in the rewards calculation and this for kudos and gamification.

Reviewer: The gamification program lets you optionally leverage pools by checking the "Use Pools" box. It means budget settings are applicable per pool rather than globally.

> For each case, you have three ways for controlling your rewarding budget :

Reviewer: ```suggestion
To control the rewards budget, pick one of the 3 policies:
```

> - **By a fixed budget:** This option allows you to set a total amount of tokens for all users rewards. This amount will be divided between the different users. The calculation will be done according to the chosen mode (use pools or not). 

Reviewer: ```suggestion
- **By a fixed budget:** Set a fixed total amount of tokens to spend in rewards. This amount of tokens will be spread between the different users in proportion of their points compared to others. If pools are used, the tokens received by each user are calculated relatively to other members of the pool rather than all users. 
```

> - **By a fixed budget per eligible member:** This mode lets you set a dynamic budget sized according to the number of active (eligible) users.  The calculation will be done according to the chosen mode (use of pools or not).

Reviewer: ```suggestion
- **By a fixed budget per eligible member:**  This is a more dynamic way where the budget is sized dynamically according to the number of eligible (having earned more than the minimal threshold) users.  Here too, actual rewards are calculated taking in account pools if the option is checked.
```

> - **By a fixed budget per point:** This mode lets you define a budget per so-called "point" (for each kudos or gamification received).

Reviewer: ```suggestion
- **By a fixed budget per point:** This is a very dynamic mode where each points earns a fixed amount of rewards. It is recommended for programs where the number of "points" is capped (like Kudos, but unlike gamification).
```

> Finally you have to save the changes using the appropriate button. You can edit this tab if you want to update the configuration using the edit button.

Reviewer: ```suggestion
Click Save to validate and apply your rewarding programs configuration.
```

Pools
~~~~~

Pools is the second tab of Reward interface. 

> As described above, this tab permits you to create pools and affect users to these pools according to their teams, roles or any other criteria you choose in order to ensure a fair distribution of rewards. 

Reviewer: ```suggestion
As described above, this tab let you create pools and assign users to them according to their teams, roles, job title or any other criteria you choose in order to ensure a fair distribution of rewards. For fair distribution, all pool members should have equal chances to compete with others. Thus, they should have access to same or similar gamified areas in the workplace.
```

> When no user have created a wallet yet you will have this interface.

Reviewer: ```suggestion
When no pool has been defined and no user have created a wallet yet, the interface will look like this
```

|image338|

> When users create their wallets, you will have a default pool containing the users that don't belong to a pool yet.

Reviewer: ```suggestion
When users create their wallets, they will be automatically assigned to the default pool until you assign them to an appropriate one.
```

|image340|

To create a pool, click on the '+' button. You only need to enter the pool's name to be able to create it. 

|image339|

You can also optionally  add a descripton, choose a space in order to add all its members to the pool and/or select a pool manager.

> There is three ways to calculate rewards for each pool members:

Reviewer: ```suggestion
You can fine tune budget per pool :
```

> - **By computing pool reward from total budget: ** When you choose this option the reward is calculated based on total reward budget.

Reviewer: ```suggestion
- **By computing pool reward from total budget: ** It's the default option. The budget attributed to this pool will be computed automatically according to the budget set in the Rewarding programs configuration
```

> - **By a total fixed budget: ** When you choose this option the reward is calculated based on a pool budget that you will have to enter. The calulation is made for all pools members even those who aren't eligible for the concerned period.

Reviewer: ```suggestion
- **By a total fixed budget: ** With this option, this pool's budget will not be taken from the global budget. Instead the pool will have its own fixed budget.
```

> - **By a fixed budget per eligible member: ** When you choose this option the reward is calculated based on a pool budget that you will have to enter. The calulation is made only for eligible members.

Reviewer: ```suggestion
- **By a fixed budget per eligible member: ** With this option, the pool has its own dynamic budget based on the number of eligible users in the pool.
```

> Finally you will have to add the appropriate users to each pool and save the action.

Reviewer: ```suggestion
Then you can add users to the pool and save.
```

It is possible to enable or disable a pool using the available option from the pool's card. If a pool is disabled at the moment of sending rewards, its members will not recieve tokens.

|image341|

|image342|

Send Rewards
~~~~~~~~~~~~~

Once everything is setup correctly, you can send the rewards from the first tab of reward interface.

|image343|

> Reward sending is allowed once the selected period of reward is passed. we can not send rewards before the end of the period conforming to the periodicity set in the configuration tab.

Reviewer: ```suggestion
Reward sending is allowed after the selected period of reward is past. You can't send rewards before the end of the period set to the periodicity set in the Configuration tab.
```

The button "Send rewards" is disabled when the current period is selected.

|image344|

From this interface, you have the following information:

- **Eligible users:** Displays the number of eligible users for the selected period.

> - **Total budget:** Displays the total budget that will be distributed for the selected period.

Reviewer: ```suggestion
- **Total budget:** Displays the total amount of tokens (to be) distributed for the selected period.
```

- **Sent Cauri:** Displays the amount of token sent if the transaction has been proceeded.

> - **Reserve fund:** Displays the amount of token in the reserve.

Reviewer: ```suggestion
- **Reserve fund:** Displays the amount of token available  in the reserve. Tokens are sent from the reserve to users. Make sure you have enough funds in the reserve before sending rewards or some users may not receive anything.
```

> - **Total kudos:** Displays the total number of sent and recieved kudos during the selected period.

Reviewer: ```suggestion
- **Total kudos:** Displays the total number of kudos received during the selected period.
```

- **Total gamification:** Displays the total number of gamification points earned during the selected period.

> You have the possibility to search for a user by his name, pool or wallet address. 

Reviewer: ```suggestion
For convenience, you can search for a user by his name, pool or wallet address. 
```
